### PR TITLE
Test TreehouseAppContent and its internal state machines

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt [hunk 1]
```diff
@@ -74,9 +73,9 @@ internal class TreehouseAppContent<A : AppService>(
   private val dispatchers: TreehouseDispatchers,
   private val appScope: CoroutineScope,
   private val eventPublisher: EventPublisher,
-  private val source: TreehouseContentSource<A>,
   private val codeListener: CodeListener,
-) : Content {
+  private val source: TreehouseContentSource<A>,
```
Comment (veyndan) [inline redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt:77]: [nitpick] Was the reordering deliberate?

File: redwood-treehouse-host/src/commonTest/kotlin/app/cash/redwood/treehouse/FakeDispatchers.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.treehouse
+
+import kotlinx.coroutines.CoroutineDispatcher
+
+class FakeDispatchers(
+  override val ui: CoroutineDispatcher,
+  override val zipline: CoroutineDispatcher,
```
Comment (veyndan) [inline redwood-treehouse-host/src/commonTest/kotlin/app/cash/redwood/treehouse/FakeDispatchers.kt:22]: As we're not using these values, perhaps remove the constructor and just override these values in the class body?

**Review by veyndan — APPROVED** [review]

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt (hunk 1 above)
Comment (swankjesse) [reply redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseAppContent.kt:77]: Yep! As a rule I try to order variables by their scope; most-singleton things first followed by least-singleton things. In this case the codeListener has a longer lifetime than the source.

(I don’t impose this on other people)

File: redwood-treehouse-host/src/commonTest/kotlin/app/cash/redwood/treehouse/FakeDispatchers.kt (hunk 2 above)
Comment (swankjesse) [reply redwood-treehouse-host/src/commonTest/kotlin/app/cash/redwood/treehouse/FakeDispatchers.kt:22]: This one’s tricky. They’re `override` values from the interface.

**Pull request merged**